src/cudadev/CUDADataFormats/SiPixelClustersCUDA.h

```diff
@@ -4,12 +4,49 @@
 #include "CUDACore/device_unique_ptr.h"
 #include "CUDACore/host_unique_ptr.h"
 #include "CUDACore/cudaCompat.h"
+#include "DataFormats/SoALayout.h"
+#include "DataFormats/SoAView.h"
 
 #include <cuda_runtime.h>
 
 class SiPixelClustersCUDA {
 public:
-  SiPixelClustersCUDA() = default;
+  GENERATE_SOA_LAYOUT(DeviceLayoutTemplate,
```

Comment:
can this be simplified using `GENERATE_SOA_LAYOUT_VIEW_AND_CONST_VIEW` ?

Reply:
Indeed, this code predates the introduction of this utility macro.